certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -565,6 +566,45 @@ def _has_certbot_redirect(self, vhost, domain):
         test_redirect_block = _test_block_from_block(_redirect_block_for_domain(domain))
         return vhost.contains_list(test_redirect_block)
 
+    def _should_break_block(self, vhost):
+        # if host has both SSL and non-SSL addrs, we should break this block!
+        return vhost.ssl and any([not addr.ssl for addr in vhost.addrs])
+
+    def _set_http_header(self, domain, header_substring):
+        """Enables header identified by header_substring on domain.
+
+        If the vhost is listening plaintextishly, separates out the relevant
+        directives into a new server block, and only add header directive to
+        HTTPS block.
+
+        :param str domain: the domain to enable header for.
+        :param str header_substring: String to uniquely identify a header.
+                        e.g. Strict-Transport-Security, Upgrade-Insecure-Requests
+        :returns: Success
+        :raises .errors.PluginError: If no viable HTTPS host can be created or
+            set with header header_substring.
+        """
+        vhost = self.choose_vhost(domain)
+        if vhost.has_header(header_substring):
+            raise errors.PluginEnhancementAlreadyPresent(
+                "Existing %s header" % (header_substring))
+
+        if vhost is None:
+            # this should never happen, since enhancements occur after
```

Comment:
We're looking to make `enhance` a verb so this comment might not be applicable forever. Change it or not, up to you.

Reply:
But this check should probably happen before we try to call `has_header` on a `None` vhost anyway.